cmd/tofu/main.go

```diff
@@ -287,15 +292,22 @@ func realMain() int {
 		return 1
 	}
 
-	// We shortcut "--version" and "-v" to just show the version
-	for _, arg := range args {
-		if arg == "-v" || arg == "-version" || arg == "--version" {
-			newArgs := make([]string, len(args)+1)
-			newArgs[0] = "version"
-			copy(newArgs[1:], args)
-			args = newArgs
-			break
-		}
+	// Set to the version command if version has been toggled
+	if _, isGlobalOptionSet := opts[optionVersion]; isGlobalOptionSet {
+		newArgs := make([]string, len(args)+1)
+		newArgs[0] = "version"
+		copy(newArgs[1:], args)
+		args = newArgs
+	}
+
+	// Attach the help option to the command args to activate help if it has been toggled
+	if _, isGlobalOptionSet := opts[optionHelp]; isGlobalOptionSet {
+		args = append(args, optionHelp)
+	}
+
+	// Attach the pedantic option to the command args to activate pedantic mode if it has been toggled
+	if _, isGlobalOptionSet := opts[optionPedantic]; isGlobalOptionSet {
+		args = append(args, optionPedantic)
 	}
```

Comment:
I think _appending_ here is a little risky because IIRC the option parsing done by many commands (using [the Go stdlib `flag` package](https://pkg.go.dev/flag)) expects to find all of the option arguments at the start of the argument slice, before any positional arguments.

That is why `mergeEnvArgs` is going to so much trouble to find a suitable place to insert the extra environment-variable-provided options, rather than just appending them all at the end:

https://github.com/opentofu/opentofu/blob/2424a632bef4933c0b7bf910a40fb59b12753e3b/cmd/tofu/main.go#L376-L418

Perhaps we could adapt that existing code to also support _this_ case of adding an extra option into the argument slice. Although the existing precedent for handling stuff like this is to pass another argument to `initCommands` above, and have that function directly set a field in the `command.Meta` object so that the `command` package code doesn't need to re-parse all this stuff:

https://github.com/opentofu/opentofu/blob/2424a632bef4933c0b7bf910a40fb59b12753e3b/cmd/tofu/main.go#L252-L256

(Notice that the `originalWd` argument is part of the behavior of the `-chdir` global option, which is most direct existing precedent for handling those directly rather than indirectly via re-parsing of the command line.)

Adapting the function that deals with the environment variables is probably the least disruptive change to what this PR is already doing, so I'm not opposed to that if it seems like that would get this done without having to significantly refactor the existing work. The main thing, I think, is to ensure that the additional options get added immediately after the subcommand, like `{"plan", "-pedantic", "whatever"}` instead of `{"plan", "whatever", "-pedantic"}`.